src/sentry/utils/circuit_breaker.py

```diff
@@ -0,0 +1,37 @@
+from django.core.cache import cache
+
+from sentry import ratelimits as ratelimiter
+from sentry.utils import metrics
+
+DEFAULT_ERROR_LIMIT = 30
+ERROR_COUNT_CACHE_KEY = lambda key: f"circuit_breaker:{key}-error-count"
+PASSTHROUGH_RATELIMIT_KEY = lambda key: f"circuit_breaker:{key}-passthrough"
+
+
+def circuit_breaker_activated(
+    key: str,
+    error_limit: int = DEFAULT_ERROR_LIMIT,
+    passthrough_data: list | None = None,
```

Comment:
I'm wondering if it's easier to reason about this if it's a dictionary. Then we can add a typed dict for it and pass it through here, so it's clear which keys are expected